TraitsExpanded/Interfaces/TraitSet.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Collections.Generic;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace TraitsExpanded
+{
+    public abstract class TraitSet
+    {
+        public virtual List<ITrait> Traits { get; set; } = new List<ITrait>();
+        public virtual string ID { get; set; }
+        public virtual int CurrentTraitIndex { get; set; } = 0;
+        public virtual NPCRestrictionEnum NPCRestriction { get; set; } = NPCRestrictionEnum.NONE;
+        public virtual bool IsActive { get; set; } = false;
+        public virtual CharacterObject CurrentCharacter { get; set; }
+
+        public virtual void Init(CharacterObject character)
+        {
+            CurrentCharacter = character;
+            Util.LogMessage($"{ID} initialized");
+        }
+
+        public virtual bool Activate()
+        {
+            var succeeded = false;
```

Comment:
under what circumstances would activating or deactivating a trait fail?

Reply:
If the character is null. I made these return bool in the miro board, but it seems that returning bool is not needed at the moment, so I will probably change it come next iteration